File: app/src/main/java/stanislav/kleinikov/exchangerate/util/BigDecimalAdapter.java
package stanislav.kleinikov.exchangerate.util;

import com.squareup.moshi.FromJson;
import com.squareup.moshi.ToJson;

import java.math.BigDecimal;

@SuppressWarnings("unused")
public class BigDecimalAdapter {
    @FromJson
    public BigDecimal fromJson(String string) {
        return new BigDecimal(string);
    }

    @ToJson
    public String toJson(BigDecimal value) {
        return value.toString();
    }
}

File: app/src/main/java/stanislav/kleinikov/exchangerate/data/adapter/BigDecimalAdapter.kt
package stanislav.kleinikov.exchangerate.data.adapter

import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson

import java.math.BigDecimal

class BigDecimalAdapter {

    @FromJson
    fun fromJson(string: String): BigDecimal = BigDecimal(string)

    @ToJson
    fun toJson(value: BigDecimal): String = value.toString()
}
